### DISCUSSION TITLE
Show Dynamic Tooltip / Popup programmatically when not bound to any Control at an arbitrary screen location

### DISCUSSION THREAD

**Discussion by fitdev** [AvaloniaUI/Avalonia#14485] [Q&A]
I know that it is possible to pre-define rich tooltips for controls in XAML or in code using attached properties. However, what if I want to show a tooltip completely dynamically from code, i.e. something like:

```cs
TooltipStaticClassOrInstance.ShowAt(PixelPointScreenLocation, Control, TimeoutTimespan); //Control here represents the Contents of the tooltip, NOT the control to which it should be bound or where it should be triggered
```

If such an API does not already exist in Avalonia. Can someone point me in the direction how I can make such an API using built-in Tooltip or Popup classes as a base perhaps.

**Comment by jp2masa** [comment]
I don't know your use case, but it sounds like you're looking for flyouts:

https://docs.avaloniaui.net/docs/reference/controls/flyouts

**Reply by fitdev** [reply]
Thank you for your suggestion. It seems that all Flyouts require a `Control` to which they will be "attached". This is not my case, as the flyout/tooltip need to be invoked entirely from code without reference to any specific control - just given a position in screen coordinates.

**Reply by timunie** [reply]
check out how FluentAvalonia TeachingTip and ContentDialog work as well as how Notifications in Avalonia  works. Also you could use a Popup. with the needed Placement.

**Reply by fitdev** [reply]
Thank you for your tips. I am wondering - could I use perhaps a regular `Window` (styled to look like a tooltip, with no chrome, etc) which I could show / hide / set position at on demand from code? Or is this approach too expensive in terms of resources?

It seems to me that thjis is going to be much easier to implement than trying to come up with a derived class from `Popup` since it still apparently requires a `Control` target (which I do not have).

**Reply by jp2masa** [reply]
If you don't have a target to attach the flyout/popup to (not even a window), creating a window is the only option, or there would be no region to render your content to.
